paf-mvp-frontend/src/utils/queue.ts

```diff
@@ -0,0 +1,64 @@
+import { Log } from '@core/log';
+
+const log = new Log('PAF', '#3bb8c3');
+
+// API for loading PAF asynchronously and run functions regardless
+// whether it gets loaded before or after their script executes.
+//
+// <script src="url/to/PAF-lib.js" async></script>
+// <script>
+//  var PAF = PAF || {};
+//  PAF.queue = PAF.queue || [];
+//  PAF.queue.push(functionToExecuteOncePrebidLoads);
+// </script>
+//
+// In asynchronous code:
+//  export const queue: CommandQueue = window.PAF.queue || [];
+//  window.PAF.queue = processCommands(queue);
+
+/** An operation executed asynchronously after the PAF-lib is loaded. */
+export type Command = () => void;
+
+/** Interface for processing every pushed commands as soon as possible.  */
+export interface ICommandProcessor {
+  push(...ops: Command[]): void;
+}
+
+/** Type for handling a duck-typing approach on 'push' function.  */
+export type CommandQueue = Command[] | ICommandProcessor;
+
+/**
+ * @param pusher Commands to process or Processor that is already in place.
+ * @returns The given processor or a new one that has just processed the given commands.
+ */
+export const processCommands = (queue: CommandQueue): ICommandProcessor => {
+  if (queue instanceof CommandProcessor) {
+    return queue;
+  }
+
+  const processor = new CommandProcessor();
+
+  if (queue && Array.isArray(queue)) {
+    processor.push(...queue);
+  }
+
+  return processor;
```

Comment:
Why do you return the processor? At first it confused me and I lost some time to understand the whole design, now I don't see what's the point of returning it.

Reply:
If you adapt the declaration of queue as I suggest in https://github.com/prebid/paf-mvp-implementation/pull/169#discussion_r877500884, this is not needed (and I insist, I think it is a better pattern to initialize it "the right way" rather than to re-affect it after a call to this method)

Reply:
Let's close comment #169 before this one.